apps/etterlatte-gyldig-soeknad/src/main/kotlin/no/nav/etterlatte/gyldigsoeknad/NySoeknadRiver.kt

```diff
@@ -44,64 +51,76 @@ internal class NySoeknadRiver(
         packet: JsonMessage,
         context: MessageContext,
     ) {
+        val soeknadId = packet.soeknadId()
+
         try {
-            logger.info("Ny søknad mottatt (id=${packet.soeknadId()})")
+            logger.info("Ny søknad mottatt (id=$soeknadId)")
 
+            val soeknad = packet.soeknad()
             val soekerFnr = packet[SoeknadInnsendt.fnrSoekerKey].textValue()
-            val soeknadType = SoeknadType.valueOf(packet[GyldigSoeknadVurdert.skjemaInfoTypeKey].textValue())
 
             val sakType =
-                when (soeknadType) {
+                when (soeknad.type) {
                     SoeknadType.BARNEPENSJON -> SakType.BARNEPENSJON
                     SoeknadType.OMSTILLINGSSTOENAD -> SakType.OMSTILLINGSSTOENAD
                 }
 
-            logger.info("Soknad ${packet.soeknadId()} er gyldig for fordeling, henter sakId for Gjenny")
-            hentSakId(soekerFnr, sakType)?.let { sakId ->
-                packet.leggPaaSakId(sakId)
-                context.publish(packet.leggPaaFordeltStatus(true).toJson())
+            val sak = finnEllerOpprettSak(soekerFnr, sakType)
+
+            if (sak == null) {
+                logger.warn("Kan ikke journalføre søknad (id=$soeknadId) uten sak. Retry kjøres automatisk...")
```

Comment:
Hmm i Hvilke tilfeller skal dette skje uten at det er en error?

Reply:
Error logges på linje 88. Warning her er bare for å ikke logge dobbel error og gjør det bittelitt tydeligere for utvikler hva som skjer. Har planer om å rydde litt mer i det når flyten har blitt forenklet og det ikke er så mange "foot guns" igjen.